### PR TITLE
CNV-11391 Removed wording that says SR-IOV network does not support live migr…

### PR DESCRIPTION
**Pull request by sjhala-ccs**
…ation.

Addresses [CNV-11391](https://issues.redhat.com/browse/CNV-11391)

OpenShift Virtualization 4.8 will support live migration on SR-IOV network connections if the `sriovLiveMigration` feature gate is enabled in the HCO CR. Changed existing note text, which said SR-IOV prevents live migration, in the following modules:

[How OpenShift Virtualization upgrades affect your cluster](https://deploy-preview-31757--osdocs.netlify.app/openshift-enterprise/latest/virt/upgrading-virt.html#how-openshift-virtualization-upgrades-affect-your-cluster)

[Storage feature matrix](https://deploy-preview-31757--osdocs.netlify.app/openshift-enterprise/latest/virt/virtual_machines/virtual_disks/virt-features-for-storage.html#virt-features-for-storage-matrix_virt-features-for-storage)

[Understanding live migration](https://deploy-preview-31757--osdocs.netlify.app/openshift-enterprise/latest/virt/live_migration/virt-live-migration.html#virt-understanding-live-migration_virt-live-migration)

[Configuring SR-IOV device for VMs](https://deploy-preview-31757--osdocs.netlify.app/openshift-enterprise/latest/virt/virtual_machines/vm_networking/virt-configuring-sriov-device-for-vms.html)

### CODE REVIEW

**Comment by ormergi** [comment]
Hi @sjenning, please keep in mind that currently SRIOV migration is feature-gated
CNV allow migration for VMs with SRIOV interfaces only when SRIOVLiveMigration feature-gate is open.

**Review by ormergi — CHANGES_REQUESTED** [review]
Hi @sjenning, 
We need to mention that SRIOV migration is feature-gated.
CNV allow migration for VMs with SRIOV interfaces only when SRIOVLiveMigration feature-gate is open.

**Comment by sjhala-ccs** [comment]
> We need to mention that SRIOV migration is feature-gated.
> CNV allow migration for VMs with SRIOV interfaces only when SRIOVLiveMigration feature-gate is open.

@ormergi What do you mean by "feature gate is open"? Is there a parameter in the `SriovNetwork` object that needs to be set?

**Comment by ormergi** [comment]
> @ormergi What do you mean by "feature gate is open"? Is there a parameter in the `SriovNetwork` object that needs to be set?

We added new feature gate for SRIOV Live Migration, the feature-gate is SRIOVLiveMigration.
When enabled:
- VM's with SRIOV network interfaces can be live migrated.
- ` virt-launcher` pod will run with `SYS_RESOURCE` capability, having this capability could place regression to virt-launcher security.
Like other feaute-gate this can be configured by hyper-converged-operator CustomResource
https://github.com/kubevirt/hyperconverged-cluster-operator/blob/master/docs/cluster-configuration.md#featuregates
https://github.com/kubevirt/hyperconverged-cluster-operator/blob/master/docs/cluster-configuration.md#sriovlivemigration-feature-gate

To provide better user experience It work in a way that if `SRIOVLiveMigration` feature-gate is open, 
`LiveMigration` feature gate will takes affect as well.
But not the counterpart; if only `LiveMigration` feature-gate is open then `SRIOVLiveMigration` will not take affect.

**Comment by sjhala-ccs** [comment]
@ormergi I have changed the content per your suggestion. Can you please review again and add LGTM to this PR if the changes look good? Thank you.

**Comment by ormergi** [comment]
Thanks you 

/lgtm

File: virt/virtual_machines/vm_networking/virt-configuring-sriov-device-for-vms.adoc [hunk 1]
```diff
@@ -10,7 +10,7 @@ This process is similar but not identical to configuring an SR-IOV device for {p
 
 [NOTE]
 ====
-Live migration is not supported for virtual machines that are attached to an SR-IOV network interface.
+Live migration is supported for virtual machines that are attached to an SR-IOV network interface only if the `sriovLiveMigration` feature gate is enabled in the HyperConverged Cluster custom resource (CR). When the `spec.featureGates.sriovLiveMigration` field is set to `true`, the `virt-launcher` pod runs with `SYS_RESOURCE` capability which might degrade its security.
```
Comment (ousleyp) [inline virt/virtual_machines/vm_networking/virt-configuring-sriov-device-for-vms.adoc:13]: The previous comment also applies here.

**Review by ousleyp — COMMENTED** [review]
LGTM, just a couple of nits/suggestions for clarity. Take 'em or leave 'em :)

**Pull request merged**